src/objective/regression_obj.cu

```diff
@@ -73,18 +73,26 @@ class RegLossObj : public ObjFunction {
           << "Number of weights should be equal to number of data points.";
     }
     auto scale_pos_weight = param_.scale_pos_weight;
-    common::Transform<>::Init(
-        [=] XGBOOST_DEVICE(size_t _idx,
+    HostDeviceVector<float> scale_pos_weight_;
+    scale_pos_weight_.Resize(1);
+    scale_pos_weight_.Fill(scale_pos_weight);
+    HostDeviceVector<int> is_null_weight_;
+    is_null_weight_.Resize(1);
+    is_null_weight_.Fill(is_null_weight);
+
+    common::Transform<>::Init([] XGBOOST_DEVICE(size_t _idx,
```

Comment:
Is it not enough to change [=] to [&] in lambda?
Also, if we need to set scalars to the function - do we really need use common::Span instead of just scalars?

Reply:
You can't pass a ref of host data to device.

Reply:
seems scalars are also not applicable, as it isn't aligned with common interface of `Transform::Eval`